src/main.rs

```diff
@@ -212,7 +235,85 @@ async fn run(cli: Cli) -> anyhow::Result<()> {
     .await?;
 
     log::info!("Done");
-    Ok(())
+    Ok(exit_code)
+}
+
+async fn run_test(test_args: TestArgs) -> anyhow::Result<i32> {
+    let TestArgs {
+        ref command,
+        upload_args,
+    } = test_args;
+    let UploadArgs {
+        ref junit_paths,
+        ref org_url_slug,
+        ref token,
+        ref repo_root,
+        ref repo_url,
+        ref repo_head_sha,
+        ref repo_head_branch,
+        ref repo_head_commit_epoch,
+        tags: _,
+        print_files: _,
+        dry_run: _,
+    } = upload_args;
+
+    let repo = BundleRepo::try_read_from_root(
+        repo_root.clone(),
+        repo_url.clone(),
+        repo_head_sha.clone(),
+        repo_head_branch.clone(),
+        repo_head_commit_epoch.clone(),
+    )?;
+
+    if junit_paths.is_empty() {
+        return Err(anyhow::anyhow!("No junit paths provided."));
+    }
+
+    let api_address = from_non_empty_or_default(
+        std::env::var("TRUNK_API_ADDRESS").ok(),
```

Comment:
Since we started using `TRUNK_API_ADDRESS` in more than one place, consider moving it to constants.